src/antares/tsgen/ts_generator.py

```diff
@@ -248,16 +348,101 @@ def _combine_failure_rates(rates1: FloatArray, rates2: FloatArray) -> None:
     rates2[mask] *= (1 - rates1[mask]) / (1 - rates2[mask])
 
 
-class ThermalDataGenerator:
+class TimeseriesGenerator:
     def __init__(self, rng: RNG = MersenneTwisterRNG(), days: int = 365) -> None:
         self.rng = rng
         self.days = days
 
-    def generate_time_series(
+    def _compare_apparent_PO(self, current_available_units: int, po_candidates: int, stock: int) -> tuple[int, int]:
```

Comment:
It could be better to rename this with `po` in lower case to fit with Python guidelines, see : https://peps.python.org/pep-0008/#function-and-variable-names

Also you could make it a function instead of a method as the parameter `self` isn't used.